pkg/idp/provider/azure/azure.go

```diff
@@ -318,3 +336,82 @@ func (a *Azure) computeAppUpdatePatch(config provider.AppConfig, app models.Appl
 
 	return appNeedsUpdate, appPatch
 }
+
+// TODO
+// improve output
+// include new service principal creation
+func (a *Azure) GetCredentialsForAuthenticatedApp(config provider.AppConfig) (string, error) {
+	ctx := context.Background()
+	app, err := a.GetApp(config.Name)
+	if err != nil {
+		if !IsNotFound(err) {
+			return "", microerror.Mask(err)
+		}
+		// Create app if it does not exist
+		app = models.NewApplication()
+		app.SetDisplayName(&config.Name)
+
+		// Set permissions from parent app
+		parentApp, err := a.GetApp(DexOperatorName)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+		if needsUpdate, patch := computePermissionsUpdatePatch(app, parentApp); needsUpdate {
+			app.SetRequiredResourceAccess(patch)
+		}
+		app, err = a.Client.Applications().Post(ctx, app, nil)
+		if err != nil {
+			return "", microerror.Maskf(requestFailedError, PrintOdataError(err))
+		}
+		a.Log.Info(fmt.Sprintf("Created %s app %s for %s in microsoft ad tenant %s", a.Type, config.Name, a.Owner, a.TenantID))
+		if app.GetAppId() == nil {
+			return "", microerror.Maskf(notFoundError, "Could not find client ID of app %s.", config.Name)
+		}
+		consentURL := getAdminConsentUrl(a.TenantID, *app.GetAppId())
+		a.Log.Info(fmt.Sprintf("Admin consent is needed. Please grant under the following URL: %s", consentURL))
+		a.Log.Info("Please be aware that it can take a while for the app to become available. Wait a moment before logging in and granting consent.")
+		err = open.Start(consentURL)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+	}
+
+	id := app.GetId()
+	if id == nil {
+		return "", microerror.Maskf(notFoundError, "Could not find ID of app %s.", config.Name)
+	}
+	secret, err := a.CreateOrUpdateSecret(*id, config, ctx, "", true)
```

Comment:
We use the existing functionality for dex-app management to create a new secret for dex-operator